docs/data_sharing_guide.md

```diff
@@ -0,0 +1,338 @@
+print(f"Group policies: {len(policies.group_policies)}")
+```
+
+## Sharing SQL Warehouse Tables
+
+### Share Tables with Users
+
+```python
+# Share a table with specific users
+response = share_table(
+    namespace="research",
+    table_name="climate_data",
+    with_users=["bob", "alice"]  # kbase usernames
+)
+
+print(f"Shared with users: {response.shared_with_users}")
+print(f"Success count: {response.success_count}")
+if response.errors:
+    print(f"Errors: {response.errors}")
+```
+
+### Share Tables with Groups
+
+```python
+# Share a table with groups
+response = share_table(
+    namespace="analytics", 
+    table_name="user_metrics",
+    with_groups=["data-science-team", "research-group"]
+)
+
+print(f"Shared with groups: {response.shared_with_groups}")
+```
+
+### Share with Both Users and Groups
+
+```python
+# Share with combination of users and groups
+response = share_table(
+    namespace="experiments",
+    table_name="results_2024",
+    with_users=["researcher1", "analyst2"],
+    with_groups=["biology-lab"]
+)
+
+print(f"Successfully shared with {response.success_count} recipients")
+```
+
+## Unsharing SQL Warehouse Tables
+
+### Remove Access from Users
+
+```python
+# Remove access from specific users
+response = unshare_table(
+    namespace="research",
+    table_name="climate_data",
+    from_users=["bob"]  # Remove bob's access, alice keeps access
+)
+
+print(f"Removed access from: {response.unshared_from_users}")
+```
+
+### Remove Access from Groups
+
+```python
+# Remove group access
+response = unshare_table(
+    namespace="analytics",
+    table_name="user_metrics", 
+    from_groups=["data-science-team"]
+)
+
+print(f"Removed group access from: {response.unshared_from_groups}")
+```
+
+### Unshare from Both Users and Groups
+
+```python
+# Unshare from both users and groups
+response = unshare_table(
+    namespace="sensitive",
+    table_name="confidential_data",
+    from_users=["bob", "alice"],
+    from_groups=["research-team", "data-scientists"]
+)
+
+print(f"Completely privatized table: {response.success_count} removals")
+```
+
+## Public and Private Table Access
+
+### Make Tables Publicly Accessible
+
+```python
+# Make a table publicly accessible to all users
+response = make_table_public(
+    namespace="public_data",
+    table_name="climate_dataset"
+)
+
+print(f"Table is now public: {response.is_public}")
+print(f"Public path: {response.path}")
+```
+
+### Make Tables Private
+
+```python
+# Remove public access and make table private again
+response = make_table_private(
+    namespace="public_data", 
+    table_name="climate_dataset"
+)
+
+print(f"Table is now private: {not response.is_public}")
+```
+
+## Managing Access Information
+
+### Check Who Has Access to Paths
+
+```python
+# Get detailed access information for any path that in your SQL warehouse
+access_info = get_path_access_info(
+    path="s3a://cdm-lake/users-general-warehouse/alice/datasets/climate-analysis"
```

Comment:
I would explain the path structure, e.g. <warehouse_prefix>/<database_name>/<table_name>

Reply:
Or better, update the function so it can take either the path or the db and table name, or make a new function that does that

Reply:
👍 added explanation of path format. For the function update - we can see how user thinks and follow another ticket. I don't think I will add function for this user guide PR.

Reply:
I would at least make a github issue for it. It's odd that all the functions that affect db permissions take the namespace and table except for the one to see the permissions

Reply:
👍 made get_table_access_info method to replace get_path_access_info

Reply:
I think we could remove it from general users. Users/Devs can still use the client to call the API.

Reply:
Users can only share their SQL warehouse path.

Reply:
Ok, I can foresee users wanting to share files in their non-sql areas but if that's not in the current plan that fine